### ISSUE TITLE
Assigning weights for each group in SNAP.train

### ISSUE THREAD

**Issue by AyeshaUlde** [materialsvirtuallab/maml#433]
Hello,

I'm trying to recreate the results in [this paper](https://pubs.acs.org/doi/10.1021/acs.jpca.9b08723) using the [data given in the mlearn repo](https://github.com/materialsvirtuallab/mlearn/tree/master/data). The [tutorial on nanoHUB](https://nanohub.org/resources/maml) assigns 10000 and 1 as the weights for energy and force respectively. The [supplementary material with the paper](https://pubs.acs.org/doi/10.1021/acs.jpca.9b08723?goto=supporting-info) gives a list of optimized hyperparameters for each group of data (eg. Energy weight of elastic group, Force weight of elastic group, etc) for each element. On using convert_docs after pooling the structures, energies and forces, the dataframe obtained does not specify the group. How can I assign the optimized weights corresponding to each specific group?

**Comment by JiQi535** [comment]
Hi @AyeshaUlde, the data weights can be set for SNAP training as in this notebook [(https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb)](https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb) 

For example, if we want to set 100 as weight for the energies of the AIMD structures, we can modify the line of `weights[df['dtype'] == 'energy'] = 100` to `weights[(df["dtype"] == "energy") & (df["group"] == "AIMD")] = 100`

**Comment by JiQi535** [comment]
The "group" information can be extracted from the training.json files. For example, if you have already extracted the "group" labels for the training structures in a list called "groups", you can add this part of information to the dataframe by `df["group"]=groups`. This is one way to have the structures and energies in the `df` correctly pointed to their "group" labels, you can try others as well.

Overall, the way to assign weights for energies, forces and stresses are already shown in the notebook [(https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb)](https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb), and the "group" labels are provided in the training.json files.

**Comment by AyeshaUlde** [comment]
> The "group" information can be extracted from the training.json files. For example, if you have already extracted the "group" labels for the training structures in a list called "groups", you can add this part of information to the dataframe by `df["group"]=groups`. This is one way to have the structures and energies in the `df` correctly pointed to their "group" labels, you can try others as well.
> 
> Overall, the way to assign weights for energies, forces and stresses are already shown in the notebook [(https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb)](https://github.com/materialsvirtuallab/maml/blob/master/notebooks/pes/snap/example.ipynb), and the "group" labels are provided in the training.json files.

Okay, got it. Thank you.
